packages/core/addon/models/table.js

```diff
@@ -54,19 +58,21 @@ export default VisualizationBase.extend(Validations, {
       metrics = get(request, 'metrics') || [],
       columns = get(this, 'metadata.columns'),
       // index column based on metricId or dimensionId
-      columnIndex = indexColumnById(columns);
+      columnIndex = indexColumnById(columns),
+      timeGrain = get(request, 'logicalTable.timeGrain.name');
 
     let dateColumn = {
       type: 'dateTime',
       attributes: { name: 'dateTime' },
       displayName: 'Date'
     };
 
-    let newColumns = [
-      dateColumn,
-      ...buildDimensionColumns(dimensions, columnIndex),
-      ...buildMetricColumns(metrics, columnIndex)
-    ];
+    let newColumns = [...buildDimensionColumns(dimensions, columnIndex), ...buildMetricColumns(metrics, columnIndex)];
+
+    //Add DateTime column when on timegrain other than All
+    if (timeGrain !== 'all') {
+      newColumns.unshift(dateColumn);
+    }
```

Comment:
Have you seen this technique?
```js
const newColumns = [
...(timeGrain !== 'all') && [dateColumn], 
...buildDimensionColumns(dimensions, columnIndex),
...buildMetricColumns(metrics, columnIndex)
];
```

Reply:
I haven't. I didn't know you could do that with the spread operator

Reply:
![Screen Shot 2019-04-01 at 3 40 42 PM](https://user-images.githubusercontent.com/23023478/55358467-1757c880-5495-11e9-8b6f-25101bc5a324.png)

Couldn't get the spread to work like that so I changed it a different way to not use `unshift`